Common/Exceptions/DllNotFoundPythonExceptionInterpreter.cs

```diff
@@ -48,8 +48,17 @@ public bool CanInterpret(Exception exception)
         public Exception Interpret(Exception exception, IExceptionInterpreter innerInterpreter)
         {
             var dnfe = (DllNotFoundException)exception;
+            
+            var startIndex = dnfe.Message.IndexOf("python");
+            var length = Math.Min(dnfe.Message.Length - startIndex, 10);
+            var dllName = dnfe.Message.Substring(startIndex, length);
+
+            length = dllName.IndexOf('\'');
```

Comment:
I believe it's possible that `DllNotFoundException` is thrown and have it not be related to python, so we should be careful here and expect strings that don't meet the pattern we expect. For example, we should handle 
```
if (startIndex < 0) // not python, emit a different message
```

In your tests, did you notice if the original message made it to the log file?

Reply:
`DllNotFoundPythonExceptionInterpreter.CanInterpret` is only true if the exception message contains `python`, so `startIndex` is always greater than `-1`.

Yes, the original message makes to the log file.
Here is the message from the user in #1746:
> 20180322 19:51:24.602 ERROR:: Loader.TryCreatePythonAlgorithm(): System.Exception: AlgorithmPythonWrapper(): The dynamic-link library for could not be found. Please visit https://github.com/QuantConnect/Lean/blob/master/Algorithm.Python/readme.md for instructions on how to enable python support in Unix python3.6

The original message is **python3.6** in the end.

Reply:
awesome, love seeing the infrastructure working hard!

Reply:
So long as we're logging the full, original message, then this works for me.